src/aspire/estimation/__init__.py

```diff
@@ -14,16 +14,24 @@
 
 
 class Estimator:
-    def __init__(self, src, basis, as_type='single', batch_size=512, preconditioner='circulant'):
+    def __init__(self, src, basis, dtype=np.float32, batch_size=512, preconditioner='circulant'):
         self.src = src
         self.basis = basis
-        self.as_type = as_type
+        self.dtype = np.dtype(dtype)
         self.batch_size = batch_size
         self.preconditioner = preconditioner
 
         self.L = src.L
         self.n = src.n
 
+        if not self.src.dtype == self.basis.dtype == self.dtype:
+            # TODO: Decide if this should simply infer from src or basis.
```

Comment:
`src` should have precedence here.

Reply:
Cool, that simplified the places we use this from explicit to pass through (inferred from src). Good stuff.